trino/trino.go

```diff
@@ -983,12 +988,12 @@ func (qr *driverRows) Close() error {
 	if qr.stmt.user != "" {
 		hs.Add(trinoUserHeader, qr.stmt.user)
 	}
-	req, err := qr.stmt.conn.newRequest("DELETE", qr.stmt.conn.baseURL+"/v1/query/"+url.PathEscape(qr.queryID), nil, hs)
+	ctx, cancel := context.WithTimeout(context.WithoutCancel(qr.ctx), DefaultCancelQueryTimeout)
```

Comment:
Whoops, `context.WithoutCancel()` was added in Go 1.21. To keep supporting older versions (>=1.17), we have to either:
1. Create a wrapper, like suggested here: https://stackoverflow.com/a/54132324
2. Create a new context and try to copy some known values, like OTEL spans and/or traces

Option 2 sounds safer. I now started wondering if `context.WithoutCancel()` would actually work with `context.WithTimeout()`.

Reply:
3 sounds good for my use as long as the copy mechanism is configurable, either the function used, or the ability to add keys that will be copied

Reply:
It would have to be similar to how the custom http client is injected, which is a bit cumbersome. I'll try to write a test for option 2.

Reply:
Since a few months passed already, I'd go back to option 1 - update minimum required Go version. Users can keep using the current driver version until they can migrate to latest Go.